### DISCUSSION TITLE
[SQLITE] how to add default column value

### DISCUSSION THREAD

**Discussion by ClaudeMa** [dbeaver/dbeaver#19757] [Q&A] [answered]
### Description

Adding a column with type TEXT(1)  with default value "C"  raise an error

[[SQLITE_ERROR] SQL error or missing database (default value of column [Column1] is not constant)
details are
[SQLITE_ERROR] SQL error or missing database (default value of column [Column1] is not constant)

Query is 
ALTER TABLE APPORTS ADD MODE_SAISIE TEXT(1) DEFAULT (C);
or
ALTER TABLE APPORTS ADD MODE_SAISIE  TEXT(1) DEFAULT ("C");
if default value is between quote
using query 
ALTER TABLE APPORTS ADD MODE_SAISIE  TEXT(1) DEFAULT **"C"**;
in sql editor work
![Capture d’écran_2023-01-04_11-04-52](https://user-images.githubusercontent.com/2773730/210531960-b89dc1ee-0c33-4e1d-81b8-850f792e34f9.png)

 

### DBeaver Version

Version 22.3.1.202212251901

### Operating System

Linux Debian 11 x64

### Database and driver

SQLITE 3 database

### Steps to reproduce

Add a column to table with the wizar

**Comment by uslss** [comment]
thanks for the report

**Comment by LonwoLonwo** [comment]
Ok, we need to figure out when we need to use parenthesis - and when not. 
https://github.com/dbeaver/dbeaver/issues/16972

**Comment by LonwoLonwo** [comment, marked as answer]
Hello @ClaudeMa

I checked your situation one more time.

You need to wrap your default STRING value into single quotes - and then it works correctly. If you have a number column and a number value - do not use quotes in this case.

![2023-04-20 13_30_16-DBeaver Ultimate 23 0 1 - NewTable_2](https://user-images.githubusercontent.com/45152336/233340608-d5df8f5d-d9f9-45c4-9fda-9ced8586eb1e.png)

![2023-04-20 13_30_28-NewTable_2 - Persist Changes](https://user-images.githubusercontent.com/45152336/233340611-a953246e-e5f8-444f-ab08-866bb246d6ce.png)
